benchmarks/01_trivial/01_aossoa.cpp

```diff
@@ -63,6 +64,24 @@ template <typename T> static void check3(T *dst, int count) {
     }
 }
 
+template <typename T> static void check2(T *dst, int count) {
```

Comment:
Looks like it was a bad idea in the first place to have `check3()` and `check4()` as separate functions (blame on me), as aos-"factor" may be passed as a parameter. Could you please refactor it to make it a single function for `check2()`/`check3()`/`check4()` by passing an extra argument to the function?

Reply:
Done